### PR TITLE
Fix git init issue

### LINKED ISSUE #36: Doesn't work properly when git just initialized
```
mkdir tmp
cd tmp
git init
touch hello
git add hello
```

Prints
```
2018/03/11 01:19:58 Failed to parse branch name for output: [## No commits yet on master]
```
Same for `gs`. `git status` works

### PR DESCRIPTION
**Pull request by zommerfelds**
Attempt to fix #36

### CODE REVIEW

File: commands/status/process_test.go [hunk 1]
```diff
@@ -238,6 +238,10 @@ var testCasesExtractBranch = []struct {
 		[]byte("## Initial commit on master"),
```
Comment (zommerfelds) [inline commands/status/process_test.go]: I'm not sure when this message gets printed. Is this for an older version of `git`? Maybe we should put a comment to describe how to replicate it in a real system.

**Comment by mroth** [comment]
Awesome! Thanks for taking this on.

File: commands/status/process_test.go (hunk 1 above)
Comment (mroth) [reply commands/status/process_test.go]: I wonder if perhaps this is in fact the same message and it just changed text in more recent versions of git. Perhaps a search through the git codebase could tell us that.

Comment (mroth) [reply commands/status/process_test.go]: Ah yeah, looks like it was changed in https://github.com/git/git/commit/4ddb1354e8d5daf5671d3d451a67d2d1e82d9b49

**Comment by zommerfelds** [comment]
Does git ever localize this output? (separate issue maybe)

**Comment by zommerfelds** [comment]
Is there anything left to be done before we can merge this?

**Comment by mroth** [comment]
> Is there anything left to be done before we can merge this?

Sorry for the delay, I was out of the country.

The one thing I was thinking is that this is a case we should probably have an integration test for as well, so we automatically catch it if the language changes again or similar.

**Comment by zommerfelds** [comment]
Good point. But actually, there is already a test case for it:
```
  Scenario: Banner shows no changes when in an unchanged git repo
    Given I am in a git repository
    When I successfully run `scmpuff status`
    Then the output should contain "No changes (working directory clean)"
```
So, currently the tests are not even passing on master with the newer git version. I created a PR below just to demonstrate this on CI:
https://github.com/mroth/scmpuff/issues/36

**Comment by mroth** [comment]
Oh great, I somehow forgot that integration test already existed.  This should be ready to be merged then. Thanks so much for your contribution @zommerfelds !

**Pull request merged**